examples/lighting-app/nuttx/build_sim.sh

```diff
@@ -0,0 +1,32 @@
+matter_dir=$(dirname "$0")/../../..
```

Comment:
Please add this as a supported platform for scripts/build/build_examples.py. Generally this is done via adding something to `scripts/build/builders/` and adding the corresponding items in `scripts/build/build/targets.py`

Scripts can do too many things. In this example it downloads and unzips things, ideally we should not do this on every build. We likely want:

- a nuttx docker file (some `integrations/docker/images/stage-2/chip-build-nuttx` directory with similar format as the rest that downloads and sets up the zip file)
- change the chip-build-vscode docker file to also include this unzipped version
- build_examples.py integration and not using a custom shell script.

Note: for docker files to be stable, you should not grab zip files from a branch because they are a moving target. They have to be versioned (use a tag or specific SHA)